web/src/components/dashboard/SidebarCardFormationNudge.test.tsx

```diff
@@ -0,0 +1,110 @@
+  generateProfileData,
+  generateSidebarCardContent,
+  generateUserData,
+} from "@/test/factories";
+import { mockPush, useMockRouter } from "@/test/mock/mockRouter";
+import {
+  currentUserData,
+  setupStatefulUserDataContext,
+  userDataWasNotUpdated,
+  WithStatefulUserData,
+} from "@/test/mock/withStatefulUserData";
+import { formationTaskId } from "@businessnjgovnavigator/shared/";
+import { UserData } from "@businessnjgovnavigator/shared/userData";
+import { fireEvent, render, screen, waitFor } from "@testing-library/react";
+
+jest.mock("@/lib/data-hooks/useUserData", () => ({ useUserData: jest.fn() }));
+jest.mock("next/router", () => ({
+  useRouter: jest.fn(),
+}));
+
+const Config = getMergedConfig();
+
+describe("<SidebarCardFormationNudge />", () => {
+  let card: SidebarCardContent;
+
+  const renderWithUserData = (userData: Partial<UserData>) => {
+    render(
+      <WithStatefulUserData initialUserData={generateUserData(userData)}>
+        <SidebarCardFormationNudge card={card} />
+      </WithStatefulUserData>
+    );
+  };
+
+  describe("when clicking funding button", () => {
+    beforeEach(() => {
+      jest.resetAllMocks();
+      useMockRouter({});
+      setupStatefulUserDataContext();
+      card = generateSidebarCardContent({ id: "formation-nudge" });
+    });
+
+    it("allows user in NEEDS_TO_FORM Phase to see modal", () => {
+      renderWithUserData({
+        profileData: generateProfileData({
+          businessPersona: "STARTING",
+        }),
+        preferences: generatePreferences({ visibleSidebarCards: ["formation-nudge"] }),
+      });
+
+      fireEvent.click(screen.getByTestId("cta-formation-nudge"));
+
+      expect(screen.getByText(Config.formationDateModal.header)).toBeInTheDocument();
+    });
+
+    it("saves formation date and updates user data with completed formation task", async () => {
+      const initialUserData = generateUserData({});
+      renderWithUserData(initialUserData);
+
+      fireEvent.click(screen.getByTestId("cta-formation-nudge"));
+
+      const input = screen.getByLabelText("Date of formation");
+      fireEvent.change(input, { target: { value: "05/2021" } });
+      fireEvent.blur(input);
+
+      fireEvent.click(screen.getByTestId("modal-button-primary"));
+
+      const updatedUserData = {
+        ...initialUserData,
+        profileData: {
+          ...initialUserData.profileData,
+          dateOfFormation: "2021-05-01",
+        },
+        taskProgress: {
+          ...initialUserData.taskProgress,
+          [formationTaskId]: "COMPLETED",
+        },
+      };
+
+      expect(currentUserData()).toEqual(updatedUserData);
+      expect(screen.queryByTestId(Config.formationDateModal.header)).not.toBeInTheDocument();
+      await waitFor(() =>
+        expect(mockPush).toHaveBeenCalledWith({ query: { fromForming: "true" } }, undefined, {
+          shallow: true,
+        })
+      );
+    });
+
+    it("allows user to click the cancel button", () => {
```

Comment:
I'm being a little nitpicky but I really want to help you practice writing descriptive test names, because it is a skill that will help guide you to writing tests more easily.

"allows" here is a little vague because to me it implies that there is a converse - is the user sometimes not "allowed" to click the button?

The outcome of this test, as written in the code, is not whether or not the user is "allowed" to click the button, but rather, what actually happens when they do so.

A more descriptive test name might read something like "it [] when the user clicks the cancel button" - where you fill in the brackets with the _outcome_ that we expect to happen.